src/fields.jl

```diff
@@ -16,11 +16,14 @@ $(TYPEDFIELDS)
 """
 struct Field1D{LX<:AbstractLocation} <: AbstractField
     "Array with the values of the field."
-    data :: Array
+    data
```

Comment:
Whyremove the type of `data`? I don't understand.

Reply:
Because when data is given by user, it is in array form. But it needs to be an offset type, because we have to apply halo points. Its type needs to be changed accordingly. I'm sure there's a better way.

Reply:
Then this should be ``::OffsetArray``?

Are you saying that the user cannot construct a field using simply an array as data? You just need to write a field constructor that will make this possible. E.g., think of what's the minimum information needed to be provided from the alongside with the `data::Array` and then make sure that constructor converts the data to the appropriate form required by `Field1D` before constructing the field. Is this clear?

Reply:
On this, before I mark it as resolved, I just wanted to point out [on some issues this could result](https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type).